specification/iotoperations/IoTOperations.Management/models/dataflows/dataflowGraphs.tsp

```diff
@@ -0,0 +1,201 @@
+
+  /** List of nodes in the dataflow graph. */
+  @OpenAPI.extension("x-ms-identifiers", #["name"])
+  nodes: DataflowGraphNode[];
+
+  /** List of connections between nodes in the dataflow graph. */
+  @OpenAPI.extension("x-ms-identifiers", #[])
+  nodeConnections: DataflowGraphNodeConnection[];
+
+  /** The provisioning state of the dataflow graph. */
+  @visibility(Lifecycle.Read)
+  provisioningState?: ProvisioningState;
+}
+
+/** DataflowGraph node properties. */
+@added(Versions.`2025-07-01-preview`)
+@discriminator("type")
+model DataflowGraphNode {
+  /** Name of the node. */
+  name: string;
+
+  /** Type of the node. */
+  type: DataflowGraphNodeType;
+}
+
+/** DataflowGraph node types. */
+@added(Versions.`2025-07-01-preview`)
+union DataflowGraphNodeType {
+  string,
+
+  /** Dataflow source node. */
+  Source,
+
+  /** Dataflow graph node. */
+  Graph,
+
+  /** Dataflow destination node. */
+  Destination,
+}
+
+/** DataflowGraph source node properties. */
+@added(Versions.`2025-07-01-preview`)
+model DataflowGraphSourceNode extends DataflowGraphNode {
+  /** Type of the source node. */
+  type: Source;
+
+  /** Source configuration. */
+  sourceSettings: DataflowGraphSourceSettings;
+}
+
+/** DataflowGraph source node settings. */
+@added(Versions.`2025-07-01-preview`)
+model DataflowGraphSourceSettings {
+  /** The endpoint reference for the source. */
+  endpointRef: string;
+
+  /** List of data sources. */
+  dataSources: string[];
+}
+
+/** DataflowGraph graph node properties. */
+@added(Versions.`2025-07-01-preview`)
+model DataflowGraphGraphNode extends DataflowGraphNode {
+  /** Type of the graph node. */
+  type: Graph;
+
+  /** Graph configuration. */
+  graphSettings: DataflowGraphNodeGraphSettings;
+}
+
+/** DataflowGraph graph node settings. */
+@added(Versions.`2025-07-01-preview`)
+model DataflowGraphNodeGraphSettings {
+  /** Reference to the registry endpoint for pulling the artifact. */
+  registryEndpointRef: string;
+
+  /** The artifact name and version to pull. */
+  artifact: string;
+
+  /** Configuration key-value pairs. */
+  @OpenAPI.extension("x-ms-identifiers", #["key"])
+  configuration?: DataflowGraphGraphNodeConfiguration[];
+}
+
+/** DataflowGraph graph node configuration. */
+@added(Versions.`2025-07-01-preview`)
+model DataflowGraphGraphNodeConfiguration {
+  /** Key of the configuration. */
+  key: string;
+
+  /** Value of the configuration. */
+  value: string;
+}
+
+/** DataflowGraph destination node properties. */
+@added(Versions.`2025-07-01-preview`)
+model DatafloGraphDestinationNode extends DataflowGraphNode {
+  /** Type of the destination node. */
+  type: Destination;
+
+  /** Destination configuration. */
+  destinationSettings: DataflowGraphDestinationNodeSettings;
+}
+
+/** DataflowGraph destination node settings. */
+@added(Versions.`2025-07-01-preview`)
+model DataflowGraphDestinationNodeSettings {
+  /** The endpoint reference for the destination. */
+  endpointRef: string;
```

Comment:
Could perhaps use a fuller documentation of how it is an 'endpointRef' as opposed to an endpoint (or uri??).